code/_onclick/item_attack.dm

```diff
@@ -123,7 +123,9 @@
 // Proximity_flag is 1 if this afterattack was called on something adjacent, in your square, or on your person.
 // Click parameters is the params string from byond Click() code, see that documentation.
 /obj/item/proc/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
-	return
+	if(is_hot(src) && target.reagents && !ismob(target) && proximity_flag)
+		to_chat(user, "<span class='notice'>You heat [target].</span>")
+		target.reagents.temperature_reagents(is_hot(src))
```

Comment:
This is kinda gross.

`is_hot` is a disgusting proc, and having everything check for this is kinda eh; this should maybe be a component one day?